src/iuse.cpp

```diff
@@ -2117,30 +2137,49 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
     rn -= mod_count * 10;                              // Other mods
 
     if( rn <= 8 ) {
-        p->add_msg_if_player(m_bad, _("You damage your %s trying to modify it!"),
-                             mod->tname().c_str());
-        if( mod->inc_damage() ) {
+        consume_mats = false;
+        consume_reqs = true;
+        consumption_modifier = 1 / 2;
+        filament_consumed = filament_required * consumption_modifier;
+        p->add_msg_if_player(m_bad, _("You damage your %s trying to modify it and also waste %d filament charges!"),
+            mod->tname().c_str(), filament_consumed);
+        if (mod->inc_damage()) {
             p->add_msg_if_player(m_bad, _("You destroy it!"));
-            p->i_rem_keep_contents( pos );
+            p->i_rem_keep_contents(pos);
         }
-        return thread_needed / 2;
     } else if( rn <= 10 ) {
-        p->add_msg_if_player( m_bad,
-                              _("You fail to modify the clothing, and you waste thread and materials.") );
-        p->consume_items( comps );
-        return thread_needed;
+        consume_mats = true;
+        consume_reqs = true;
+        consumption_modifier = 1;
+        filament_consumed = filament_required * consumption_modifier;
+        p->add_msg_if_player(m_bad,
+            _("You fail to modify the clothing, and you waste filament (%d charges) and materials."), filament_consumed);
     } else if( rn <= 14 ) {
-        p->add_msg_if_player( m_mixed, _("You modify your %s, but waste a lot of thread."),
-                              mod->tname().c_str() );
-        p->consume_items( comps );
-        mod->item_tags.insert( the_mod );
-        return thread_needed;
+        consume_mats = true;
+        consume_reqs = true;
+        consumption_modifier = 2;
+        filament_consumed = filament_required * consumption_modifier;
+        mod->item_tags.insert(the_mod);
+        p->add_msg_if_player(m_mixed, _("You modify your %s, but waste a lot of filament (%d charges)."), mod->tname().c_str(), filament_consumed);
```

Comment:
In every section, you recalculate the number just to print what you used. Would be cleaner if you had it once in consumption section. It's fine for it to be split into 2 messages, for example
"You use %d charges"
"You fail to modify %s"